pageserver/src/error.rs

```diff
@@ -0,0 +1,61 @@
+//!
+//! The `error` module provides a general [`Error`] struct representing different types of
+//! encountered errors when running the page server.
+//!
+
+use thiserror::Error;
+
+#[derive(Error)]
+pub enum Error {
+    #[error("Postgres IO error: {0:?}")]
+    PostgresIO(#[from] PostgresIOError),
```

Comment:
I created `PostgresIO` variant to represent read/write errors when interacting while PG clients. atm, it only includes the `ConnectionReset` error. Rust doc mentions that "[std::io::Error is] the error type for I/O operations of the Read, Write, Seek, and associated traits.", so I think it also includes network i/o operations.

Reply:
Ok. How about `NetworkIOError`? I don't like the Postgres part here because as a server, we don't really know what the client is. For the 'basebackup' command for example, the client is actually the `zenith_ctl` utility that runs in the compute node before Postgres itself.

Reply:
Or just `NetworkError`